addon/commands/wrap-selection.ts

```diff
@@ -20,11 +27,16 @@ export function wrapSelection(nodeType: NodeType): Command {
       if (!nodeRange) {
         return false;
       }
-      const wrappers = findWrapping(nodeRange, nodeType);
+      let wrappers = findWrapping(nodeRange, nodeType);
       if (!wrappers) {
         return false;
       }
       if (dispatch) {
+        if (attrsFromWrapped) {
+          const attrs = attrsFromWrapped(nodeRange);
+          const newWrap = findWrapping(nodeRange, nodeType, attrs);
+          wrappers = newWrap ?? wrappers;
+        }
```

Comment:
Would it not be better to move this part before the dispatch and calculating the wrapping only once? (based on if `attrsFromWrapped` have been passed or not).

Reply:
My thinking was that since `attrsFromWrapped` is passed in, we don't have any way to know if it's heavy (or has side-effects, etc.), so we should only run it if we're actually `dispatch`ing a result rather than checking if the command can run. This is probably over thinking it though and it would be better being easier to read.

Reply:
I can see that, but I suppose it is then our responsibility to pass in a function that is not too heavy.